File: software/examples/ruby/example_threshold.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_distance_ir_v2'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your Distance IR Bricklet 2.0

ipcon = IPConnection.new # Create IP connection
dir = BrickletDistanceIRV2.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Register distance callback
dir.register_callback(BrickletDistanceIRV2::CALLBACK_DISTANCE) do |distance|
  puts "Distance: #{distance/10.0} cm"
end

# Configure threshold for distance "smaller than 30 cm"
# with a debounce period of 1s (1000ms)
dir.set_distance_callback_configuration 1000, false, '<', 30*10, 0

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect

File: software/examples/perl/example_threshold.pl
#!/usr/bin/perl

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickletDistanceIRV2;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your Distance IR Bricklet 2.0

# Callback subroutine for distance callback
sub cb_distance
{
    my ($distance) = @_;

    print "Distance: " . $distance/10.0 . " cm\n";
}

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $dir = Tinkerforge::BrickletDistanceIRV2->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Register distance callback to subroutine cb_distance
$dir->register_callback($dir->CALLBACK_DISTANCE, 'cb_distance');

# Configure threshold for distance "smaller than 30 cm"
# with a debounce period of 1s (1000ms)
$dir->set_distance_callback_configuration(1000, 0, '<', 30*10, 0);

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();
